src/io/flutter/run/daemon/FlutterAppManager.java

```diff
@@ -178,7 +182,7 @@ void stopApp(@NotNull FlutterApp app) {
       // streams are closed which immediately terminates the process.
       CompletableFuture
         .completedFuture(sendCommand(app.getController(), cmd))
-        .thenApply(this::waitForResponse)
+        .thenApply((Method method) -> waitForResponse(method, 300))
```

Comment:
300ms - it seems like a good max time to wait, in terms of the stop button being responsive.